npe2/_command_registry.py

```diff
@@ -27,9 +27,6 @@ def register(self, id: str, command: Callable) -> PDisposable:
         if not callable(command):
             raise TypeError(f"Cannot register non-callable command: {command}")
 
-        # TODO: validate argumemnts and type constraints
-        # possibly wrap command in a type validator?
```

Comment:
I'm in favor of handling this w #29 right now; the cli tool can do some initial validation to type check.